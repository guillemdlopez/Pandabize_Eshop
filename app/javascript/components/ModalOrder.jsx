import React, { useContext } from "react";
import PropTypes from "prop-types";
import { useHistory } from "react-router-dom";
import { userContext } from "../context/userContext";

const ModalOrder = ({
  content,
  name,
  photo,
  price,
  setSuccessMsg,
  successMsg,
  displaySuccess,
}) => {
  const { user } = useContext(userContext);
  const history = useHistory();
  console.log(typeof displaySuccess);

  const goToDashboard = () => {
    history.push("/dashboard");
  };

  const goToHomePage = () => {
    history.push("/");
  };

  return (
    <div className="modal-order animate__animated animate__fadeIn">
      <div className="text-center">
        <i className="far fa-check-circle fa-5x mb-3 text-success text-center"></i>
        <p className="mb-4">
          <strong>{user}</strong>, {content}
        </p>
      </div>
      <div className="order-card d-flex align-items-center">
        <img src={photo} className="order-card-img" />
        <div className="order-card-content ml-4">
          <ul className="block-list-left mb-0">
            <li className="mb-0">
              <strong>Bicycle:</strong> {name}
            </li>
            <li className="mb-0">
              <strong>Price:</strong>{" "}
              <span className="price-tag">{price.toFixed(2)} â‚¬</span>
            </li>
          </ul>
        </div>
      </div>
      <button
        className="btn btn-success btn-block mt-5"
        onClick={goToDashboard}
      >
        Go to dashboard
      </button>
      <button className="btn-customized" onClick={goToHomePage}>
        Home
      </button>
    </div>
  );
};

ModalOrder.propTypes = {
  content: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  setSuccessMsg: PropTypes.func.isRequired,
  successMsg: PropTypes.object.isRequired,
  displaySuccess: PropTypes.bool.isRequired,
};

export default ModalOrder;
