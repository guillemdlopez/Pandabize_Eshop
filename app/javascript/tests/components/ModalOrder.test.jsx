import { mount } from "enzyme";
import React from "react";
import { MemoryRouter } from "react-router";
import ModalOrder from "../../components/ModalOrder";
import { userContext } from "../../context/userContext";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("Testing the component <ModalOrder/>", () => {
  const userInfo = {
    user: "Guillem",
    userId: 1,
  };

  const modalProps = {
    content: "your order has been succesfully completed!",
    name: "Black Panther 5000",
    photo: "img.jpg",
    price: 25,
  };
  const wrapper = mount(
    <MemoryRouter>
      <userContext.Provider value={userInfo}>
        <ModalOrder {...modalProps} />
      </userContext.Provider>
    </MemoryRouter>
  );
  test("should render the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should containa a message with the name of the current user", () => {
    const { content } = modalProps;
    const { user } = userInfo;
    expect(wrapper.find("p").text()).toBe(`${user}, ${content}`);
  });

  test("should contain the name of the bicycle", () => {
    const { name: bicycle } = modalProps;
    expect(wrapper.find("li").at(0).text()).toBe(`Bicycle: ${bicycle}`);
  });

  test("should contain the price of the bicycle", () => {
    const { price } = modalProps;
    expect(wrapper.find("li").at(1).text()).toBe(
      `Price: ${price.toFixed(2)} â‚¬`
    );
  });

  test("should contain a button to go to the dashboard", () => {
    const btnDashboard = wrapper.find("button").at(0);
    expect(btnDashboard.exists()).toBe(true);
    expect(btnDashboard.text()).toBe("Go to dashboard");
  });

  test("should contain a button to go to the home page", () => {
    const btnHome = wrapper.find("button").at(1);
    expect(btnHome.exists()).toBe(true);
    expect(btnHome.text()).toBe("Home");
  });

  test("should redirect to the Dashboard when the user clicks on the button", () => {
    const btnDashboard = wrapper.find("button").at(0);
    btnDashboard.simulate("click");

    expect(mockHistoryPush).toBeCalledWith("/dashboard");
  });

  test("should redirect to the Home page when the user clicks on the button", () => {
    const btnHome = wrapper.find("button").at(1);
    btnHome.simulate("click");

    expect(mockHistoryPush).toHaveBeenCalledWith("/");
  });
});
