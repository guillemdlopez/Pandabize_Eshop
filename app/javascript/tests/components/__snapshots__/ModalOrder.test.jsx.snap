// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the component <ModalOrder/> should render the component correctly 1`] = `
<div
  className="modal-order animate__animated animate__fadeIn"
>
  <div
    className="text-center"
  >
    <i
      className="far fa-check-circle fa-5x mb-3 text-success text-center"
    />
    <p
      className="mb-4"
    >
      <strong>
        Guillem
      </strong>
      , 
      your order has been succesfully completed!
    </p>
  </div>
  <div
    className="order-card d-flex align-items-center"
  >
    <img
      className="order-card-img"
      src="img.jpg"
    />
    <div
      className="order-card-content ml-4"
    >
      <ul
        className="block-list-left mb-0"
      >
        <li
          className="mb-0"
        >
          <strong>
            Bicycle:
          </strong>
           
          Black Panther 5000
        </li>
        <li
          className="mb-0"
        >
          <strong>
            Price:
          </strong>
           
          <span
            className="price-tag"
          >
            25.00
             â‚¬
          </span>
        </li>
      </ul>
    </div>
  </div>
  <button
    className="btn btn-success btn-block mt-5"
    onClick={[Function]}
  >
    Go to dashboard
  </button>
  <button
    className="btn-customized"
    onClick={[Function]}
  >
    Home
  </button>
</div>
`;
